### DISCUSSION TITLE
How to implement a setTimeout close function on the Dialog component?

### DISCUSSION THREAD

**Discussion by arnelamo** [shadcn-ui/ui#1943] [Q&A]
Hi, everyone. 

I was wondering if anyone here have an idea of how to best implement an setTimeout function that closes the Dialog component after a set number of seconds?

I currently use the useEffect + useState (open, setOpen) hooks to close it, on the open/onOpenChange props of the Root component, after a set number of seconds, but I would be happy to get tips for a better solution.

**Comment by onurhan1337** [comment]
While there are various approaches to handling this, the structure I've used in my projects is quite similar. Notably, I haven't received any negative feedback regarding performance or readability issues with this approach. Hence, I find it to be a suitable and effective solution for this particular situation.
